hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -410,6 +413,14 @@ private boolean initializeFromFilesystem(String initializationTime, List<Metadat
             }
             fileGroupCountAndRecordsPair = initializePartitionStatsIndex(partitionInfoList);
             break;
+          case SECONDARY_INDEX:
+            Set<String> secondaryIndexPartitionsToInit = getSecondaryIndexPartitionsToInit();
+            if (secondaryIndexPartitionsToInit.isEmpty()) {
+              continue;
+            }
+            ValidationUtils.checkState(secondaryIndexPartitionsToInit.size() == 1, "Only one secondary index at a time is supported for now");
```

Comment:
not sure why we have the number limit.

Reply:
While we can create multiple secondary indexes using configs, but can only create one at a time using sql due to limitation of sql grammar. To keep the user experience consistent, I have added this validation. This will be removed later on after we enahcne the sql grammar and parser to support creating multiples indexes at a time.

Reply:
> but can only create one at a time using sql due to limitation of sql grammar.

Reasonable, but it also looks like we only allow one index for MDT bootstrap, is this expected?

Reply:
Then what about the other ones, they would never got a chance to bootstrap right

Reply:
I see your point. Instead of failing, we should still continue to bootstrap other index types. I will change the logic a bit.